src/java/org/apache/cassandra/cql3/QueryOptions.java

```diff
@@ -53,6 +54,10 @@
 
     public static final CBCodec<QueryOptions> codec = new Codec();
 
+    private static final int UNSET_NOWINSEC = Integer.MIN_VALUE;
+    // Guava uint counterpart to the above for serialization purposes. This imposes a year 2186 max value
+    private static final int UNSET_NOWINSEC_GUAVA_UINT = UnsignedInteger.MAX_VALUE.intValue() - 1;
```

Comment:
Why do we subtract one?

Reply:
I think this might be a copy-paste case form the `LEGACY_OVERFLOWED_DELETION_TIME` elsewhere. I'll change and run CI to make sure it so indeed as this is quite a self-contained case, I don't see any other interfering values getting in.

Reply:
Confirmed by CI